rest/src/main/kotlin/com/gitlab/kordlib/rest/request/KtorRequestHandler.kt

```diff
@@ -80,20 +76,25 @@ class KtorRequestHandler(
             parameters.appendAll(request.parameters)
         }
 
-        request.body?.let {
-            val body = parser.encodeToString(it.strategy, it.body)
-            logger.debug { request.logString(body) }
-            when (request) {
-                is MultipartRequest<*, *> -> {
-                    headers.append("payload_json", body)
-                    this.body = MultiPartFormDataContent(request.data)
+        when (request) {
+            is JsonRequest -> {
+                request.body?.let {
+                    val json = parser.encodeToString(it.strategy, it.body)
+                    logger.debug { request.logString(json) }
+                    this.body = TextContent(json, io.ktor.http.ContentType.Application.Json)
```

Comment:
We can do an early return here and remove the let scope, makes it easier to read. But we can change this later

Reply:
I decided against it initially because you can't return to a when, and returning out of the method early seemed like a source of bugs when we're nested this deeply. I could wrap this branch in a `run` though and return to that.